contrib/vagrant/start.sh

```diff
@@ -576,36 +576,91 @@ function vboxnet_addr_finder(){
     vboxnet_add_ipv4 "${vboxnetname}" "${IPV4_BASE_ADDR_NFS}" "255.255.255.0"
 }
 
-vboxnet_addr_finder
 
-ipv6_public_workers_addrs=()
+function createVm(){
+    vboxnet_addr_finder
 
-split_ipv4 ipv4_array "${MASTER_IPV4}"
-MASTER_IPV6="${IPV6_INTERNAL_CIDR}$(printf '%02X' ${ipv4_array[3]})"
+    ipv6_public_workers_addrs=()
 
-set_reload_if_vm_exists
+    split_ipv4 ipv4_array "${MASTER_IPV4}"
+    MASTER_IPV6="${IPV6_INTERNAL_CIDR}$(printf '%02X' ${ipv4_array[3]})"
 
-create_master
-create_workers
-set_vagrant_env
-create_k8s_config
+    set_reload_if_vm_exists
 
-cd "${dir}/../.."
+    create_master
+    create_workers
+    set_vagrant_env
+    create_k8s_config
 
-PROVISION_ARGS=""
-if [ -n "${NO_PROVISION}" ]; then
-    PROVISION_ARGS="--no-provision"
-fi
-if [ -n "${RELOAD}" ]; then
-    vagrant reload $PROVISION_ARGS $1
-elif [ -n "${PROVISION}" ]; then
-    vagrant provision $1
-else
-    vagrant up $PROVISION_ARGS $1
-    if [ "$?" -eq "0" -a -n "${K8S}" ]; then
-        host_port=$(vagrant port --guest 6443 k8s1)
-        vagrant ssh k8s1 -- cat /home/vagrant/.kube/config | sed "s;server:.*:6443;server: https://k8s1:$host_port;g" > vagrant.kubeconfig
-        echo "Add '127.0.0.1 k8s1' to your /etc/hosts to use vagrant.kubeconfig file for kubectl"
+    cd "${dir}/../.."
+
+    PROVISION_ARGS=""
+    if [ -n "${NO_PROVISION}" ]; then
+        PROVISION_ARGS="--no-provision"
     fi
-fi
+    if [ -n "${RELOAD}" ]; then
+        vagrant reload $PROVISION_ARGS $1
+    elif [ -n "${PROVISION}" ]; then
+        vagrant provision $1
+    else
+        vagrant up $PROVISION_ARGS $1
+        if [ "$?" -eq "0" -a -n "${K8S}" ]; then
+            host_port=$(vagrant port --guest 6443 k8s1)
+            vagrant ssh k8s1 -- cat /home/vagrant/.kube/config | sed "s;server:.*:6443;server: https://k8s1:$host_port;g" > vagrant.kubeconfig
+            echo "Add '127.0.0.1 k8s1' to your /etc/hosts to use vagrant.kubeconfig file for kubectl"
+        fi
+    fi
+}
 
+#get number of running VM(s)
+runningVm=$(VBoxManage list runningvms | awk 'END{ print NR }')
+VMName=$(VBoxManage list runningvms | awk 'NR==1{print $1}' |  cut -d "\"" -f 2)
+if [ "$VMName" ]; then
+    echo
+	echo "Detected running VMs that might cause conflict"
+	echo
+    VBoxManage list runningvms
+
+	#option to stop, delete VM(s) or ignore and continue
+    echo
```

Comment:
Could you fix up all indentations in this file? It looks like there's a mix of tabulations and spaces.